docs/proposals/device-crd.md

```diff
@@ -0,0 +1,650 @@
+	// Required: Limit number of registers to read/write.
+	Limit          int64              `json:"limit,omitempty"`
+	// The scale to convert raw property data into final units.
+	// Defaults to 1.0
+	// +optional
+	Scale          float64            `json:"scale,omitempty"`
+	// Indicates whether the high and low byte swapped.
+	// Defaults to false.
+	// +optional
+	IsSwap         bool               `json:"isSwap,omitempty"`
+	// Indicates whether the high and low register swapped.
+	// Defaults to false.
+	// +optional
+	IsRegisterSwap bool               `json:"isRegisterSwap,omitempty"`
+}
+
+type ModbusRegisterType string
+
+const (
+	ModbusRegisterTypeCoilRegister          ModbusRegisterType = "CoilRegister"
+	ModbusRegisterTypeDiscreteInputRegister ModbusRegisterType = "DiscreteInputRegister"
+	ModbusRegisterTypeInputRegister         ModbusRegisterType = "InputRegister"
+	ModbusRegisterTypeHoldingRegister       ModbusRegisterType = "HoldingRegister"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DeviceModel is the Schema for the device model API
+// +k8s:openapi-gen=true
+type DeviceModel struct {
+	metav1.TypeMeta      `json:",inline"`
+	metav1.ObjectMeta    `json:"metadata,omitempty"`
+
+	Spec DeviceModelSpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DeviceModelList contains a list of DeviceModel
+type DeviceModelList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DeviceModel `json:"items"`
+}
+```
+
+### Device model sample
+```yaml
+apiVersion: devices.kubeedge.io/v1alpha1
+kind: DeviceModel
+metadata:
+  labels:
+    description: 'TI Simplelink SensorTag Device Model'
+    manufacturer: 'Texas Instruments'
+    model: CC2650
+  name: sensor-tag-model
+spec:
+  properties:
+  - name: temperature
+    description: temperature in degree celsius
+    int:
+      accessMode: ReadOnly
+      maximum: 100
+      unit: Degree Celsius
+  - name: temperature-enable
+    description: enable data collection of temperature sensor
+    string:
+      accessMode: ReadWrite
+      defaultValue: OFF
+  propertyVisitors:
+  - propertyName: temperature
+    modbus:
+      register: CoilRegister
+      offset: 2
+      limit: 1
+      scale: 1
+      isSwap: true
+      isRegisterSwap: true
+   - propertyName: temperature-enable
+      modbus:
+        register: DiscreteInputRegister
+        offset: 2
+        limit: 1
+        scale: 1
+        isSwap: true
+        isRegisterSwap: true
+```
+
+The device model shown above describes the temperature property of a sensor. Access modes describe whether the property is read-only or can be written to.
+Property visitors provide details like how to access device properties. In the above example , the visitor is for Modbus protocol and describes the register to read data , the offset to apply etc.
```

Comment:
Please remove all the whitespaces before ",":`s/\ ,/,/g`

Reply:
fixed.